internal/resources/google/network_egress_internal.go

```diff
@@ -0,0 +1,508 @@
+	usageFiltersData := r.getEgressUsageFiltersData()
+	defaultAPIRegionName := r.getEgressAPIRegionName()
+	serviceName := r.getEgressAPIServiceName()
+
+	resource := &schema.Resource{
+		Name:           r.Address,
+		CostComponents: []*schema.CostComponent{},
+	}
+
+	// Same continent
+	var quantity *decimal.Decimal
+	if r.SameContinent != nil {
+		quantity = decimalPtr(decimal.NewFromFloat(*r.SameContinent))
+	}
+	resource.CostComponents = append(resource.CostComponents, &schema.CostComponent{
+		Name:            fmt.Sprintf("%s in same continent", r.PrefixName),
+		Unit:            "GB",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: quantity,
+		ProductFilter: &schema.ProductFilter{
+			VendorName: strPtr("gcp"),
+			Region:     strPtr("global"),
+			Service:    strPtr("Cloud Storage"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "description", Value: strPtr("Inter-region GCP Storage egress within EU")},
+			},
+		},
+	})
+
+	for _, regData := range regionsData {
+		usageKey := regData.usageKey
+		usage := GetFloatFieldValueByUsageTag(usageKey, *r)
+		newCostComponents := stepPricingHelper(usage, usageFiltersData, regData, defaultAPIRegionName, serviceName)
+		resource.CostComponents = append(resource.CostComponents, newCostComponents...)
+	}
+
+	return resource
+}
+
+func (r *ComputeVPNGatewayNetworkEgressUsage) BuildResource() *schema.Resource {
+	regionsData := r.getEgressRegionsData()
+	usageFiltersData := r.getEgressUsageFiltersData()
+	defaultAPIRegionName := r.getEgressAPIRegionName()
+	serviceName := r.getEgressAPIServiceName()
+
+	resource := &schema.Resource{
+		Name:           r.Address,
+		CostComponents: []*schema.CostComponent{},
+	}
+
+	for _, regData := range regionsData {
+		usageKey := regData.usageKey
+		usage := GetFloatFieldValueByUsageTag(usageKey, *r)
+		newCostComponents := stepPricingHelper(usage, usageFiltersData, regData, defaultAPIRegionName, serviceName)
+		resource.CostComponents = append(resource.CostComponents, newCostComponents...)
+	}
+
+	return resource
+}
+
+func (r *ComputeExternalVPNGatewayNetworkEgressUsage) BuildResource() *schema.Resource {
+	regionsData := r.getEgressRegionsData()
+	usageFiltersData := r.getEgressUsageFiltersData()
+	defaultAPIRegionName := r.getEgressAPIRegionName()
+	serviceName := r.getEgressAPIServiceName()
+
+	resource := &schema.Resource{
+		Name:           r.Address,
+		CostComponents: []*schema.CostComponent{},
+	}
+
+	for _, regData := range regionsData {
+		usageKey := regData.usageKey
+		usage := GetFloatFieldValueByUsageTag(usageKey, *r)
+		newCostComponents := stepPricingHelper(usage, usageFiltersData, regData, defaultAPIRegionName, serviceName)
+		resource.CostComponents = append(resource.CostComponents, newCostComponents...)
+	}
+
+	return resource
+}
+
+func stepPricingHelper(usage float64, usageFiltersData []*egressRegionUsageFilterData, regData *egressRegionData, defaultAPIRegionName, serviceName *string) []*schema.CostComponent {
```

Comment:
```suggestion
func egressStepPricingHelper(usage float64, usageFiltersData []*egressRegionUsageFilterData, regData *egressRegionData, defaultAPIRegionName, serviceName string) []*schema.CostComponent {
```

1. I'd make the name more explicit since the function is global in this package
2. I'd suggest making making the region and service strings, since we're just wrapping them in a ptr to pass them in here which seems pointless.

Reply:
Makes total sense